vol_object_test.c

```diff
@@ -5266,16 +5312,41 @@ test_object_visit(void)
         }
         PART_END(H5Ovisit_create_order_decreasing);
 
+        PART_BEGIN(H5Ovisit_group)
+        {
+            TESTING_2("H5Ovisit on a group");
+
+            if (H5Ovisit3(group_id3, H5_INDEX_CRT_ORDER, H5_ITER_INC, object_visit_simple_callback, &i,
+                          H5O_INFO_ALL) < 0) {
+                H5_FAILED();
+                HDprintf("    H5Ovisit on a group failed!\n");
+                PART_ERROR(H5Ovisit_group);
+            }
```

Comment:
Should add a check here to make sure the value of `i` is as expected after the visit call